src/main/java/com/odysseusinc/arachne/portal/util/SubmissionHelper.java

```diff
@@ -421,6 +432,94 @@ public void updateExtendInfo(Submission submission) {
         }
     }
 
+    private class PathwaySubmissionExtendInfoStrategy extends SubmissionExtendInfoAnalyzeStrategy {
+
+        @Override
+        public void updateExtendInfo(Submission submission) {
```

Comment:
this method is too long, can you split it into a few? like `getPathwayCodes` etc